lib/msal-angular/docs/v2-docs/configuration.md

```diff
@@ -270,3 +271,231 @@ fetch('/assets/configuration.json')
   }
 }
 ```
+
+## Dynamic configurations using Factory Providers and APP_INITIALIZER
+
+To dynamically configure MSAL Angular, you can use the Factory Providers with APP_INITIALIZER
+
+`src/app/config.service.ts`
+```typescript
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpBackend } from '@angular/common/http';
+import { map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ConfigService {
+
+  private settings: any;
+  private http: HttpClient;
+  constructor(private readonly httpHandler: HttpBackend) {
+    this.http = new HttpClient(httpHandler);
+  }
+
+  init(endpoint: string): Promise<boolean> {
+    return new Promise<boolean>((resolve, reject) => {
+      this.http.get(endpoint).pipe(map(res => res))
+        .subscribe(value => {
+          this.settings = value;
+          resolve(true);
+        },
+        (error) => {
+          reject(error);
+        });
+    });
+  }
+
+  getSettings(key?: string | Array<string>): any {
+    if (!key || (Array.isArray(key) && !key[0])) {
+      return this.settings;
+    }
+
+    if (!Array.isArray(key)) {
+      key = key.split('.');
+    }
+
+    let result = key.reduce((acc: any, current: string) => acc && acc[current], this.settings);
+
+    return result;
+  }
+}
+```
+
+`src/app/msal-config-dynamic.module.ts`
+```typescript
+import { InjectionToken, NgModule, APP_INITIALIZER } from '@angular/core';
+import { IPublicClientApplication, PublicClientApplication, 
+    InteractionType, BrowserCacheLocation, LogLevel } from '@azure/msal-browser';
```

Comment:
```suggestion
    LogLevel } from '@azure/msal-browser';
```
No unused imports. Imports on same line if possible (we will be changing our other configuration methods to also do this)